### ISSUE TITLE
MSBuild /p:OutputPath is not being set with a trailing backlash

### ISSUE THREAD

**Issue by Cawifre** [fsharp/FAKE#869]
As of commit 537757392cad0582c2d7a24d3f7f4ce1b95fbe4d, the OutputPath parameter shifted from being guaranteed to always have a trailing backslash to being guaranteed to never have a trailing backslash.

[Microsoft.Common.CurrentVersion.targets](http://referencesource.microsoft.com/#MSBuildFiles/C/ProgramFiles%28x86%29/MSBuild/14.0/bin_/Microsoft.Common.CurrentVersion.targets,719) expects `$(OutputPath)` and `$(OutDir)` to have trailing backslashes to allow for concatenation. It attempts to make sure that a trailing backslash is added, but setting a property on the command line [circumvents these attempts](http://blogs.msdn.com/b/msbuild/archive/2006/10/05/_2f00_p-property-values-are-immutable-_2d00_-_2800_sort-of_2900__2e00__2e00__2e00__2e00_.aspx).

This causes issue when concatenation is done to build nested paths (for instance, [the default path for publishing application manifests becomes a sibling of the output directory instead of a child of the output directory](http://referencesource.microsoft.com/#MSBuildFiles/C/ProgramFiles%28x86%29/MSBuild/14.0/bin_/Microsoft.Common.CurrentVersion.targets,435)).

Simply adding a single trailing backslash (e.g. `/p:OutputPath="C:\MyApp\build\"`) causes a much more serious error in which the single backslash escapes the closing quote and pulls following command line arguments into the property value.

The correct command line argument for setting `$(OutputPath)` should be `/p:OutputPath="C:\MyApp\build\\"`

**Comment by dsyme** [comment]
@Cawifre Is this still an issue?

**Comment by Cawifre** [comment]
@dsyme Yes.

**Comment by Meidor** [comment]
I'm also facing this issue with the XamarinBuildHelpers but I think they use MSBuild in the end. I see that the xbuild command that it generates misses a trailing backspace and adding one to the command fixes my problems. Is there any known workaround for this problem?

**Comment by Cawifre** [comment]
@Meidor this is the workaround I use:  
```
let buildDir = @".\build\"
let expandOutputPath = FullName >> trimSeparator >> sprintf @"%s\\"
"MyApp.sln"
    |> MSBuild "" "Rebuild" [("OutputPath", expandOutputPath buildDir); ("Configuration", "Release")]
    |> Log "MyApp-Output: "
```

I've not used the XamarinBuildHelpers, but you may be able to use a similar technique.

**Comment by SimonCropp** [comment]
can confirm this is still an issue https://github.com/Fody/Fody/issues/290

**Comment by matthid** [comment]
Fixed with #1905 in rc011 thanks to @JarnoNijboer
